doc: add validation section in the TOML (#124)
AdRoll/baker: content/en/docs/How Tos/pipeline_configuration.md

defines a structure of the records with two fields: `foo` as first element and `bar` as second.

#### Validation configuration

> The `[validation]` section contains one or more field names with a regular expression.

Reviewer: ```suggestion
The `[validation]` section contains one or more field names each of which is associated with a regular expression.
```

If the validation section is specified Baker automatically generates a validation function, 

> which checks that each input record satisfied the provided regular expression. In the case, 

Reviewer: Shouldn't it be?
```suggestion
which checks that each input record satisfy the provided regular expression. In the case, 
```

> one or more field does not match the whole record is thrown away.

Reviewer: ```suggestion
one or more field does not match the whole record is discarded.
```

To make an example:

```toml
[validation]
foo = "^\w+$"
bar = "[0-9]+"
```

defines that `foo` field must be a not empty word and `bar` field must contain a number. 
In this case, a valid record could be:

| foo           | bar       |
|:-------------:|:---------:|
| `hello_world` | `hello23` |

> The regular expression reference could be found at [golang.org/s/re2syntax](https://golang.org/s/re2syntax)

Reviewer: :+1:

defines a structure of the records with two fields: `foo` as first element and `bar` as second.

#### Validation configuration

The `[validation]` section is an optional configuration that contains one or more field names each 
of which is associated with a regular expression. 
If the validation section is specified Baker automatically generates a validation function, 
which checks that each input record satisfy the provided regular expression. 

> In the case, one or more field does not match the whole record is discarded. 

Reviewer: Maybe a bit clearer:
```suggestion
The record is discarded at the first field that doesn't match its associated regular expression. 
```

> The user could choose to not provide record validation at all or to implements a more sophisticated 

Reviewer: ```suggestion
The user could choose to not provide record validation at all or to implement a more sophisticated 
```

> validation function using a go function specified in the [Components](/docs/how-tos/baker_components/#validate).

Reviewer: ```suggestion
validation function using a Go function specified in the [Components](/docs/how-tos/baker_components/#validate) struct.
```
